sprockets/mixins/avro_publisher/__init__.py

```diff
@@ -8,51 +8,164 @@
     ``http://my-schema-repository/avro/%(name)s.avsc``
 
 Take note also of the required configurations in sprockets.mixins.amqp
+
 """
+import io
 import logging
 
-LOGGER = logging.getLogger(__name__)
-
 try:
-    from sprockets.mixins import amqp
-
-    from sprockets.mixins.avro_publisher.mixins import (PublishingMixin,
-                                                        SchemaFetchError)
+    from sprockets.mixins import amqp, http
+    from tornado import gen
+    import fastavro
+except ImportError:  # pragma: nocover
+    amqp, fastavro, gen, http = object(), object(), object(), object()
+    amqp.PublishingMixin, http.HTTPClientMixin = object(), object()
```

Comment:
could you not just set all of these to one instance of None for some reason? Like
     amqp, fastavro, gen, http = None